tests/compile-test.rs

```diff
@@ -74,12 +74,16 @@ fn run_mode(mode: &str, dir: PathBuf) {
     compiletest::run_tests(&cfg);
 }
 
-#[warn(clippy::identity_conversion)]
-fn run_ui_toml_tests(config: &compiletest::Config, mut tests: Vec<TestDescAndFn>) -> Result<bool, io::Error> {
+#[allow(clippy::identity_conversion)]
```

Comment:
This line should be removed completely (sometime). It's a bug that this lint started to trigger and therefore we set it to `warn` instead of `deny` for the dogfood test. #3913 

If this bug still exists (which I think is the case) this line should stay unchanged.

Reply:
It still exists :)

Reply:
Remember, compile-test.rs is a different crate, so `allow()`s applied elsewhere will not help here.